app/javascript/components/App.js

```diff
@@ -1,65 +1,87 @@
-import React from "react"
-import{
+import React, { Component } from 'react'
+import {
   BrowserRouter as Router, Route, Switch
 } from 'react-router-dom'
 
 import Header from './components/Header'
 import Footer from './components/Footer'
 
 import DevTeam from './pages/DevTeam'
-import EmergencyContactsIndex from './pages/EmergencyContactsIndex'
+import EmergencyContactsIndex from './components/EmergencyContactsIndex'
 import Faq from './pages/Faq'
 import Home from './pages/Home'
 import LearnMore from './pages/LearnMore'
+import NotFound from './pages/NotFound'
 import MyEmergencyContactsEdit from './pages/MyEmergencyContactsEdit'
-import MyEmergencyContactsIndex from './pages/MyEmergencyContactsIndex'
+import MyEmergencyContactsIndex from './components/MyEmergencyContactsIndex'
 import MyUserProfile from './pages/MyUserProfile'
 import MyUserProfileEdit from './pages/MyUserProfileEdit'
 import UserProfile from './pages/UserProfile'
 
-class App extends React.Component {
+import mockPersonalInfo from './mockPersonalInfo.js'
+import mockEmergencyContact from './mockEmergencyContact.js'
+
+
+class App extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      personalInfo: mockPersonalInfo
+      personalInfo: mockPersonalInfo,
+      emergencyContact: mockEmergencyContact
     }
   }
   render() {
     const {
       logged_in,
       sign_in_route,
-      sign_out_route
+      sign_up_route,
+      sign_out_route,
+      current_user
     } = this.props
     return (
       <Router>
         <Header />
+
         <Switch>
-          { /* Unprotected Routes */ }
-          <Route
-            path="/userprofileindex"
-            render={ (props) => <UserProfile info={ this.state.personalInfo }/>}
-            >
-          </Route>
+          { /* Unprotected Routes */}
 
+          { /*Home*/}
+          <Route exact path="/" component={Home} />
 
-          { /* Protected Routes */ }
-        </Switch>
-      <Home />
-      <LearnMore />
-      <DevTeam />
-      <Faq />
+          {/*Only Index We'll Have is Emergency Contact*/}
 
-        { logged_in &&
+          { /*Show for Public Profile is difficult to code right now because we need to understand our routes better, so I'll Work on Emergency Contacts First*/}
+          {/*<Route
+            path="/profileindex/:id"
+            render={(props) => <UserProfile personalInfo={this.state.personalInfo} />}
+          />*/}
+
+          {/*Not Found*/}
+          <Route component={NotFound} />
+
+          { /* Protected Routes */}
+        </Switch>
+        <Home />
+        <LearnMore />
```

Comment:
I think ```<Home />``` and ```<LearnMore />``` were supposed to be removed. The ```<Route>``` tags below also need to be inside the ```<switch />```
The Router should have the form: 
```
<Router>
  <Switch>
    <Route />
    <Route />
  </Switch>
</Router>
```

Reply:
I see what you mean. It looks like the App.js component that you're looking at is outdated. On our local machines, when we pull directly from the branch react-components, the changes that you have suggested have already been made.